static/js/sentry.ts

```diff
@@ -0,0 +1,48 @@
+import * as Sentry from "@sentry/browser";
+import {HttpClient as HttpClientIntegration} from "@sentry/integrations";
+import {BrowserTracing} from "@sentry/tracing";
+import _ from "lodash";
+
+import {page_params} from "./page_params";
+
+if (page_params.server_sentry_dsn) {
+    const url_regex = new RegExp("^" + _.escapeRegExp(page_params.realm_uri) + "/");
+    Sentry.init({
+        dsn: page_params.server_sentry_dsn,
+        environment: page_params.development_environment ? "development" : "production",
+
+        release: "zulip-server@" + ZULIP_VERSION,
+        integrations: [
+            new BrowserTracing({
+                tracePropagationTargets: [url_regex],
+            }),
+            new HttpClientIntegration({
+                failedRequestStatusCodes: [500, 502, 503, 504],
+                failedRequestTargets: [url_regex],
+            }),
+        ],
+        allowUrls: [url_regex, page_params.webpack_public_path],
+        sampleRate: 1,
+        tracesSampleRate: 1,
+        initialScope: {
+            tags: {
+                realm: page_params.realm_string_id,
+            },
+            user: {
+                id: page_params.user_id?.toString(),
+                realm: page_params.realm_string_id,
+                role: page_params.is_owner
+                    ? "Organization owner"
+                    : page_params.is_admin
+                    ? "Organization administrator"
+                    : page_params.is_moderator
+                    ? "Moderator"
+                    : page_params.is_guest
+                    ? "Guest"
+                    : page_params.is_spectator
+                    ? "Spectator"
+                    : "Member",
```

Comment:
You might be able to use `settings_config.user_role_values` for this lookup with less duplication... though it's kinda the converse of the map you want.

Reply:
`page_params` doesn't ship a user role value for spectators.  We also want to load this with as few dependencies as possible, so that it init's even if there are failures in other files.  I'm not sure pulling in the 800+ lines of `settings_config.ts` is worth it here.

Reply:
Yeah, I think that's fine; we can always optimize later if we see a better opportunity; this one seems fairly messy.